src/scripts/main.js

```diff
@@ -1,5 +1,32 @@
 'use strict';
 
-document.addEventListener('click', (e) => {
+const wall = document.querySelector('.wall');
+
+const spider = document.querySelector('.spider');
+
+const wallParam = wall.getBoundingClientRect();
+
+wall.addEventListener('click', (e) => {
   // write code here
+  let left = Math.min(
+    e.clientX - wall.offsetLeft - spider.clientWidth / 2 - wall.clientLeft,
+    wallParam.right -
+      wall.offsetLeft -
+      2 * wall.clientLeft -
+      spider.clientWidth,
+  );
```

Comment:
The calculation for 'left' ensures that the spider does not move outside the right boundary of the wall. However, consider checking if 'e.clientX' is less than 'wall.offsetLeft' to prevent the spider from moving outside the left boundary.